File: 2024/helper/InputHelper.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;

public class InputHelper {
    private final String separator = System.getProperty("file.separator");

    public ArrayList<String> readLines(boolean testing, String day){
        ArrayList<String> lines = new ArrayList<>();
        try{
            String file = "inputFiles" + separator + day + separator + (testing ? "example" : "input") + ".txt";
            File myObj = new File(file);
            Scanner myReader = new Scanner(myObj);
            while (myReader.hasNextLine()){
                String line = myReader.nextLine();
                lines.add(line);
            }
        } catch (FileNotFoundException e){
            System.out.println("An error occurred.");
            e.printStackTrace();
            return null;
        }

        return lines;
    }
}

File: 2025/helper/InputHelper.cs
namespace AdventOfCode.helper;

public class InputHelper
{
    public List<string> ReadLines(Boolean testing, String day)
    {
        var fileName = $"inputFiles{Path.DirectorySeparatorChar}{day}{Path.DirectorySeparatorChar}{(testing ? "example" : "input")}.txt";
        var lines = new List<string>();
        using (StreamReader reader = new StreamReader(fileName))
        {
            string line;
            while ((line = reader.ReadLine()) is not null)
            {
                lines.Add(line);
            }
        }

        return lines;
    }   
}